lib/internal/Magento/Framework/Controller/Test/Unit/Result/RedirectTest.php

```diff
@@ -67,16 +70,44 @@ public function testSetPath()
             $this->returnValue($params)
         );
         $this->assertInstanceOf(
-            \Magento\Framework\Controller\Result\Redirect::class,
+            Redirect::class,
             $this->redirect->setPath($path, $params)
         );
     }
 
-    public function testRender()
+    public function testRenderDataProvider()
     {
-        $this->response->expects($this->once())->method('setRedirect');
+        return [
+            ['http://test.com', null, 302],
+            ['http://test.com', 0, 302],
+            ['http://test.com', '', 302],
+            ['http://test.com', 301, 301],
+            ['http://test.com', 302, 302],
+            ['http://test.com', 303, 303],
+            ['http://test.com', 200, 302],
+            ['http://test.com', 404, 302],
+            ['http://test.com', 503, 302],
```

Comment:
Too much cases as to me for such a tiny method.

First of all move URL out of data provider. Secondly, it's enough to check
```
null, 302
302, 302
303, 303 
```